### DISCUSSION TITLE
git workflows documentation

### DISCUSSION THREAD

**Discussion by leamas** [opencpn-manuals/main#10] [General]
Obviously, we will eventually need to document the workflow(s) for plugin developers in the manual.

However,  for the core manual developers I think the traditional INSTALL and README files works better. They are readily available in the directory they are used, which is far better than a manual page somewhere. There is a example in the development manual [here](https://github.com/opencpn-manuals/development/blob/main/INSTALL.adoc)

Thoughts?

**Comment by Rasbats** [comment]
Fine for me as well. Perhaps a link in the manual. I like a quick reference for achieving something.

**Comment by rgleason** [comment]
I think that is fine to use Install and Readme to help us all, however I do agree that we will need a more comprehensive page or document for the entire workflow and process. 

Indeed, I think starting and writing this page will help clarify the process. 

It might even help Jon and me do something useful, as things have been moving around here!

**Reply by leamas** [reply]
Look into https://github.com/opencpn-manuals/development/blob/main/INSTALL.adoc and https://github.com/opencpn-manuals/development/blob/main/WORKFLOWS.adoc. These are a starter...

**Reply by rgleason** [reply]
Thank you Alec this looks like very useful instructions. I think I understand it now.

In terms of getting setup to edit, should I abandon my repository setup of  ocpn-manuals  and ocpn-dev-manual and just clone the  opencpn-manuals   development repository in order to edit for the development manual?  In particular, Testplugin? 
Or should we have a separate source like you & mike are using for ShipDriver

I don't believe I need the plugins repository because the sources are in each plugin repository (I have a number of them in my repository from Mike and those are the ones I would normally edit) and we should follow your instructions to update those.

**Reply by leamas** [reply]
Basically, just clone the opencpn-manuals stuff for now and keep you old setup for reference. 

As I see it, there is no need for a separate module like that for shipdriver. That there is one for shipdriver is just a matter of history, the site existed and could be used more or less as-is when we started this work. 

The advantage of having a separate module would be that you could build a separate testplugin manual. But this is could already be done in testplugin/manual if you want to work with Jon.
